### PR TITLE
feat: add provider api token to all linode calls

### PR DESCRIPTION
**Pull request by MasterPtato**

### CODE REVIEW

**Comment by MasterPtato** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/613?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#613** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/613?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#599** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/599?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#598** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/598?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#382** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/382?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#381** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/381?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#380** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/380?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#300** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/300?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#373** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/373?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#377** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/377?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#597** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/597?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MasterPtato and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by AngelOnFira — APPROVED** [review]

**Review by AngelOnFira — APPROVED** [review]

File: svc/pkg/linode/standalone/gc/src/lib.rs [hunk 1]
```diff
@@ -40,7 +40,8 @@ pub async fn run_from_env(pools: rivet_pools::Pools) -> GlobalResult<()> {
 	);
 
 	// Build HTTP client
-	let client = util_linode::Client::new_with_headers(headers).await?;
+	let api_token = util::env::read_secret(&["linode", "token"]).await?;
```
Comment (NathanFlurry) [inline svc/pkg/linode/standalone/gc/src/lib.rs:43]: Won't this only run for the main cluster? Don't we need to iterate over all clusters?

Comment (NathanFlurry) [inline svc/pkg/linode/standalone/gc/src/lib.rs:43]: Won't this only run for the main cluster? Don't we need to iterate over all clusters?

File: svc/pkg/linode/util/src/lib.rs [hunk 2]
```diff
@@ -32,7 +31,7 @@ impl Client {
 		})
 	}
 
-	pub async fn new_with_headers(mut headers: header::HeaderMap) -> GlobalResult<Self> {
+	pub async fn new_with_headers(api_token: &str, mut headers: header::HeaderMap) -> GlobalResult<Self> {
 		let api_token = util::env::read_secret(&["linode", "token"]).await?;
```
Comment (NathanFlurry) [inline svc/pkg/linode/util/src/lib.rs:35]: This overrides the api token passed in

Comment (NathanFlurry) [reply svc/pkg/linode/util/src/lib.rs:35]: rust warning would've caught this, we need to clean these up :/

File: svc/pkg/linode/util/src/lib.rs [hunk 3]
```diff
@@ -16,8 +16,7 @@ pub struct Client {
 }
 
 impl Client {
-	pub async fn new() -> GlobalResult<Self> {
-		let api_token = util::env::read_secret(&["linode", "token"]).await?;
+	pub async fn new(api_token: &str) -> GlobalResult<Self> {
```
Comment (NathanFlurry) [inline svc/pkg/linode/util/src/lib.rs:19]: We're writing a lot of conditionals. Can we just pass in "override api token" as an optional and if not exists, read the default token?

**Review by NathanFlurry — CHANGES_REQUESTED** [review]

**Review by NathanFlurry — APPROVED** [review]

**Pull request merged**